src/main/java/com/iota/iri/service/transactionpruning/jobs/MilestonePrunerJob.java

```diff
@@ -97,6 +99,11 @@ private MilestonePrunerJob(int startingIndex, int currentIndex, int targetIndex)
         }
     }
 
+    @Override
+    public void setSpentAddressesService(SpentAddressesService spentAddressesService) {
+        //Does nothing. Confirmed transactions spent are persisted during the local snapshot.
```

Comment:
this sounds a bit fragile. maybe only validate they are in the list? (i.e. query the spent addresses when you see a negative balance and throw is it fails?)

Reply:
Throwing an exception so it will stop pruning?
Actually it is a good idea

If there is a problem we want the operator to be annoyed and not be able to prune